### ISSUE TITLE
Jwt_optional and POST method

### ISSUE THREAD

**Issue by johaven** [vimalloc/flask-jwt-extended#129]
I'm facing an issue since i updated from 3.3.4 to 3.7.0.
In 3.3.4 this kind of route worked without any problems:
```python
@app.route('/partially-protected', methods=['GET', 'POST'])
@jwt_optional
def my_route():
...
```
Since 3.7.0 during a post request the CSRF token is required.
Is it intentional ? Is it useful to protect this kind of route when auth is optional ?

**Comment by vimalloc** [comment]
That is not intentional. I'll look at what's going on and getting a fix in place later tonight (and add a unit test so that does not happen again). Sorry about that!

**Comment by vimalloc** [comment]
Fix released in 3.7.2. Cheers 👍
